lib/v8debugapi.js

```diff
@@ -384,10 +384,25 @@ module.exports.create = function(logger_, config_, fileStats_) {
   // }
 
   function findScripts(scriptPath) {
+    // Use repository relative mapping if present
+    if (config.appPathRelativeToRepository) {
+      var candidate = scriptPath.replace(config.appPathRelativeToRepository,
+        config.workingDirectory);
+      // There should be no ambiguity resolution if project root is provided
+      return fileStats[candidate] ? [ candidate ] : [];
+    }
+    // Next try to match path
     var regexp = pathToRegExp(scriptPath);
     var matches = Object.keys(fileStats).filter(function(script) {
       return regexp.test(script);
     });
+    // Finally look for files with the same name regardless of path
+    if (matches.length === 0) {
+      scriptPath = path.basename(scriptPath);
```

Comment:
Another option would be to peel one directory at a time to see if we can get a unique match. For example, I may have `Server/app/index.js` and `Server/foo/index.js`. Just looking for `index.js` will find duplicates, even though we ought to be able to do better here.